source/type/type.cpp

```diff
@@ -395,6 +396,26 @@ static bool concerns_base_at_period(const VehicleJourney& vj,
     return intersect;
 }
 
+
+static bool is_useless(const nt::VehicleJourney& vj) {
+    // a vj is useless if all it's validity pattern are empty
+    for (const auto l_vj: vj.validity_patterns) {
+        if (! l_vj.second->empty()) { return false; }
+    }
+    return true;
+}
+
+template <typename C>
+static typename C::iterator erase_vj_from_list(const nt::VehicleJourney* vj, C& vjs) {
+    auto it = boost::range::find(vjs, vj);
+    if (it == vjs.end()) {
+        LOG4CPLUS_WARN(log4cplus::Logger::getInstance("logger"),
+                       "impossible to find the vj " << vj->uri << " in the list, something is strange");
+        return it;
+    }
+    return vjs.erase(it);
```

Comment:
hmm... dangerous....

..., erasing elements in positions other than the vector end causes the container to **RELOCATE** all the elements after the segment erased to their new positions. 


```
v.erase( std::remove( vjs.begin(), vjs.end(), vj ), vjs.end() ); 
```